modules/http/src/main/java/org/mule/module/http/internal/request/DefaultHttpRequesterConfig.java

```diff
@@ -111,7 +113,15 @@ public void initialise() throws InitialisationException
                 .setOwnerName(name)
                 .build();
 
-        httpClient = new GrizzlyHttpClient(configuration);
+        GrizzlyHttpClientFactory httpClientFactory = muleContext.getRegistry().get(HTTP_CLIENT_FACTORY_NAME);
```

Comment:
Are you sure that this registry entry will always be available when this object is constructed. From what I remember the registry bootstrap stuff happens in the initialize phase.